packages/devtools_app/lib/src/service/editor/api_classes.dart

```diff
@@ -0,0 +1,418 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Known kinds of events that may come from the editor.
+///
+/// This list is not guaranteed to match actual events from any given editor as
+/// the editor might not implement all functionality or may be a future version
+/// running against an older version of this code/DevTools.
+enum EditorEventKind {
+  // Device.
+  deviceAdded,
+  deviceRemoved,
+  deviceChanged,
+  deviceSelected,
+
+  // Debug Session.
+  debugSessionStarted,
+  debugSessionChanged,
+  debugSessionStopped,
+}
+
+/// Constants for all fields used in JSON maps to avoid literal strings that
+/// may have typos sprinkled throughout the API classes.
+abstract class _Field {
+  static const category = 'category';
+  static const debuggerType = 'debuggerType';
+  static const debugSession = 'debugSession';
+  static const debugSessionId = 'debugSessionId';
+  static const device = 'device';
+  static const deviceId = 'deviceId';
+  static const emulator = 'emulator';
+  static const emulatorId = 'emulatorId';
+  static const ephemeral = 'ephemeral';
+  static const flutterDeviceId = 'flutterDeviceId';
+  static const flutterMode = 'flutterMode';
+  static const id = 'id';
+  static const name = 'name';
+  static const platform = 'platform';
+  static const platformType = 'platformType';
+  static const projectRootPath = 'projectRootPath';
+  static const supported = 'supported';
+  static const vmServiceUri = 'vmServiceUri';
+}
+
+/// A base class for all known events that an editor can produce.
+///
+/// The set of subclasses is not guaranteed to match actual events from any
+/// given editor as the editor might not implement all functionality or may be a
+/// future version running against an older version of this code/DevTools.
+sealed class EditorEvent {
```

Comment:
can you mixin `Serializable`? and then you can remove the abstract override of toJson here?

https://github.com/flutter/devtools/blob/master/packages/devtools_shared/lib/src/utils/serialization.dart/#L11

Reply:
Done! Any objection to me changing `dynamic` to `Object?` in that mixin? I dislike `dynamic` because it's easy to pass where it shouldn't go, but `Object?` prevents that :-)

Reply:
+1 to change to `Object?`.

Reply:
Done in 7199e0d98